cmd/channel/distributed/controller/main.go

```diff
@@ -34,5 +37,7 @@ func main() {
 	defer kafkasecret.Shutdown()
 
 	// Create The SharedMain Instance With The Various Controllers
-	sharedmain.Main(constants.ControllerComponentName, kafkachannel.NewController, kafkasecret.NewController)
+	ctx := signals.NewContext()
+	sharedmain.MainWithContext(controller.WithResyncPeriod(ctx, env.GetEnvironmentOrDie(ctx).ResyncPeriod),
+		constants.ControllerComponentName, kafkachannel.NewController, kafkasecret.NewController)
```

Comment:
Can you pull the controller.WithResyncPeriod() out onto it's own line or combine with ctx line above rather than nest inside MainWithContext() call (readability)?

Reply:
Sure, no problem

Reply:
Sure, no problem